### DISCUSSION TITLE
Serial number uniqueness WITH casing

### DISCUSSION THREAD

**Discussion by moseisleydk** [netbox-community/netbox#18165] [Help Wanted!] [answered]
I have implemented: https://github.com/netbox-community/netbox/issues/9187

Code:

`class UniqueSerial(CustomValidator):

    def validate(self, instance, *args, **kwargs):
        from dcim.models import Device

        if instance.serial and Device.objects.exclude(pk=instance.pk).filter(serial=instance.serial).exists():
            self.fail("The Serial number already exists in NetBox", field='serial')`

But, this does not prevent "AAA" and "aaa" as 2 serial numbers.

I tried:

`class UniqueSerial(CustomValidator):

    def validate(self, instance, *args, **kwargs):
        from dcim.models import Device

        if instance.serial and Device.objects.exclude(pk=instance.pk).filter(serial.lower()=instance.serial.lower()).exists():
            self.fail("The Serial number already exists in NetBox", field='serial')
`

This fails with:

`File "/etc/netbox/config/configuration.py", line 329
    if instance.serial and Device.objects.exclude(pk=[instance.pk](http://instance.pk/)).filter(serial.lower()=instance.serial.lower()).exists():
                                                                         ^^^^^^^^^^^^^^^
SyntaxError: expression cannot contain assignment, perhaps you meant "=="?`

and 

`class UniqueSerial(CustomValidator):

    def validate(self, instance, *args, **kwargs):
        from dcim.models import Device

        if instance.serial and Device.objects.exclude(pk=instance.pk).filter(serial.lower()==instance.serial.lower()).exists():
            self.fail("The Serial number already exists in NetBox", field='serial')
`

Also fails.

How do I add "lower" part to the code correctly (python noob...)

**Comment by jeremystretch** [comment, marked as answer]
You can use a [Django filter lookup](https://docs.djangoproject.com/en/5.1/ref/models/querysets/#iexact) to perform a case-insensitive match. This should work:

```
.filter(serial__iexact=instance.serial)
```

**Reply by moseisleydk** [reply]
Worked like a charm, thanks @jeremystretch  !!
